input/product_link.txt

```diff
@@ -0,0 +1,5 @@
+http://automationpractice.com/index.php?id_product=3&controller=product
+http://automationpractice.com/index.php?id_product=4&controller=product
+http://automationpractice.com/index.php?id_product=5&controller=product
+
```

Comment:
add one link, which leads to non-product page.
add `tryTo()` to scenario to skip such pages without fail.